src/rpc_server.rs

```diff
@@ -14,24 +14,24 @@
 //! use tarpc::client;
 //! use tarpc::context;
 //!
-//! # tokio_test::block_on(async {
-//!
+//! # #[tokio::main]
+//! # async fn main() -> Result<(), Box<dyn std::error::Error>>{
```

Comment:
are you using main() instead of tokio_test::block_on() in order to return errors, or for some other reason?    If so, that seems reasonable, but let's just use anyhow::Result<()>, which is already in scope as `Result`.   So it should work to just:

```suggestion
//! # async fn main() -> Result<()> {
```